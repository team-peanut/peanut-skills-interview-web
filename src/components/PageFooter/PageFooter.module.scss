@use "../../styles/mediaQueries";

.wrapper {
  background-color: var(--color-footer-background);
  color: var(--color-footer-text);
  width: 100%;

  > .container {
    margin: 0 auto;
    width: 100%;

    a {
      text-decoration: none;
      color: white;
    }
    @include mediaQueries.tablet {
      display: grid;
      grid-template-areas:
        "install social social"
        "contacts links links"
        "copyright language language";
      grid-template-columns: 1fr 1fr 1fr;
    }

    @include mediaQueries.desktop {
      grid-template-areas:
        "install social social"
        "contacts links ."
        "language links copyright";
      grid-template-rows: auto 1fr 77px;
      padding-bottom: 40px;
      width: var(--breakpoint-desktop);
    }

    > section {
      padding: 40px 45px;

      &:not(:last-child) {
        border-bottom: 1px solid rgba(var(--color-footer-text-rgb), 0.3);
      }

      @include mediaQueries.tablet {
        &:nth-last-child(2) {
          border-bottom: none;
        }
      }

      @include mediaQueries.desktop {
        border-bottom: 0;

        &:not(:last-child) {
          border-bottom: 0;
        }
      }

      ul {
        list-style-type: none;
        margin: 0;
        padding: 0;
        > li {
          margin: 0;
        }
      }
      .button {
        border: 0;
        background-color: white;
        border-radius: 100px;
      }

      &.install {
        display: flex;
        gap: 16px;
        grid-area: install;
        justify-content: center;

        @include mediaQueries.tablet {
          justify-content: flex-start;
          justify-self: unset;
        }

        > a {
          .button {
            height: 42px;
            width: 120px;

            @include mediaQueries.tablet {
              border-radius: 100%;
              height: 50px;
              padding: 0;
              position: relative;
              width: 50px;
            }

            .buttonIOS {
              height: 22px;
              transform: translateY(-2px);
              width: 18px;

              @include mediaQueries.tablet {
                left: 50%;
                position: absolute;
                top: 50%;
                transform: translate(-50%, -50%);
              }
            }

            .buttonAndroid {
              height: 22px;
              transform: translateY(-2px);
              width: 19px;

              @include mediaQueries.tablet {
                left: 50%;
                position: absolute;
                top: 50%;
                transform: translate(-50%, -50%);
              }
            }
          }
        }
      }

      &.social {
        grid-area: social;

        @include mediaQueries.tablet {
          align-items: center;
          display: flex;
          justify-content: flex-end;
          justify-self: unset;
        }

        > ul {
          display: flex;
          flex-direction: row;
          gap: 40px;
          justify-content: center;
          > li {
            a {
              img {
                display: block;
              }
            }
          }
        }
      }

      &.contacts {
        grid-area: contacts;
        padding-bottom: 40px;
        padding-top: 40px;

        @include mediaQueries.desktop {
          align-items: flex-start;
          display: flex;
          padding-bottom: 40px;
        }

        > ul {
          display: flex;
          flex-direction: column;
          gap: 30px;
          > li {
            color: var(--color-footer-highlighted-text);
            font-size: 15px;
            line-height: 24px;

            > a {
              color: var(--color-footer-text);
              display: block;
            }
          }
        }
      }

      &.links {
        grid-area: links;
        padding-bottom: 22px;
        padding-top: 22px;

        @include mediaQueries.desktop {
          align-items: flex-start;
          display: flex;
        }

        > div {
          @include mediaQueries.tablet {
            columns: 2;
            justify-content: flex-end;
            justify-self: unset;
          }

          > ul {
            align-self: flex-end;
            display: flex;
            flex-direction: column;
            > li {
              > a {
                display: block;
                font-size: 16px;
                padding: 17px 0;
                white-space: nowrap;
              }
            }
          }
        }
      }

      &.language {
        grid-area: language;

        @include mediaQueries.tablet {
          align-items: center;
          display: flex;
          justify-self: end;
        }

        @include mediaQueries.desktop {
          align-items: flex-end;
          display: flex;
          justify-content: flex;
          justify-self: unset;
        }

        > ul {
          display: flex;
          flex-direction: row;
          gap: 40px;
          justify-content: center;

          @include mediaQueries.desktop {
            gap: 20px;
          }

          &:before {
            align-self: stretch;
            border: 1px solid lightgrey;
            content: "";
          }

          > li {
            font-size: 18px;
            line-height: 20px;
            padding-top: 7px;
            text-align: center;
            &:first-child {
              order: -1;
            }

            @include mediaQueries.desktop {
              font-size: 16px;
            }

            > a {
              &.selected {
                color: var(--color-footer-highlighted-text);
              }
            }
          }
        }
      }

      &.copyright {
        grid-area: copyright;

        @include mediaQueries.tablet {
          align-items: center;
          display: flex;
        }

        @include mediaQueries.desktop {
          justify-content: flex-end;
          padding-top: 20px;
        }

        > div {
          background-image: url("./assets/logo.svg");
          background-position: 0 50%;
          background-repeat: no-repeat;
          background-size: 40px 22px;
          line-height: 20px;
          margin: 0 auto;
          min-height: 22px;
          padding-left: 48px;
          padding-top: 3px;
          width: fit-content;

          @include mediaQueries.tablet {
            line-height: 18px;
            margin: 0;
          }

          @include mediaQueries.desktop {
            font-size: 16px;
            height: 52px;
            line-height: 48px;
          }
        }
      }
    }
  }
}
