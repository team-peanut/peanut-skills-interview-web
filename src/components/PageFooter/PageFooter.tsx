import React from "react";

import IconApple from "./assets/apple.svg";
import IconAndroid from "./assets/robot.svg";
import IconFacebook from "./assets/facebook.svg";
import imageTikTok from "./assets/tiktok.svg";
import IconTwitter from "./assets/twitter.svg";
import IconPinterest from "./assets/pinterest.svg";
import IconInstagram from "./assets/instagram.svg";


import styles from "./PageFooter.module.scss";

const PageFooter = () => {

  return (
    <div className={styles.wrapper} data-testid="page-footer">
      <footer className={styles.container}>
        <section className={styles.install}>
            <button className={styles.button}>
              <img
                src={IconApple}
                className={styles.buttonIOS}
              />
            </button>
          
            <button className={styles.button}>
              <img
                src={IconAndroid}
                className={styles.buttonAndroid}
              />
            </button>
        </section>

        <section className={styles.social}>
          <ul>
            <li>
              <a
                href="#"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={IconInstagram}
                  alt="Instagram"
                  width="22"
                  height="22"
                />
              </a>
            </li>
            <li>
              <a
                href="#"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img src={IconFacebook} alt="Facebook" width="22" height="22" />
              </a>
            </li>

            <li>
              <a
                href="#"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img src={IconTwitter} alt="Twitter" width="27" height="22" />
              </a>
            </li>
            <li>
              <a
                href="#"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img src={imageTikTok} alt="TikTok" width="18" height="22" />
              </a>
            </li>
            <li>
              <a
                href="#"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={IconPinterest}
                  alt="Pinterest"
                  width="22"
                  height="22"
                />
              </a>
            </li>
          </ul>
        </section>

        
        <section className={styles.contacts}>
          <ul>
            <li>
              <a href="#">
                Advertising Enquiries
              </a>
            </li>
            <li>
              <a href="#">
                Press Enquiries
              </a>
            </li>
            <li>
              <a href="#">
                Something else?
              </a>
            </li>
          </ul>
        </section>
        
        <section className={styles.links}>
          <div>
            <ul>
              <li>
                <a href="#">
                  About us
                </a>
              </li>
              <li>
                <a href="#">
                  Blog
                </a>
              </li>
              <li>
                <a href="#">
                  Medical Review
                </a>
              </li>

              <li>
                <a href="#">
                  Guidelines
                </a>
              </li>

            
              <li>
                <a href="#">
                  Offers
                </a>
              </li>
            
            
              <li>
                <a href="#">
                  Start Her
                </a>
              </li>
            

              <li>
                <a href="#">
                    Shop
                </a>
              </li>

              <li>
                <a href="#">
                  Peanut News
                </a>
              </li>

              <li>
                <a href="#">
                  Careers
                </a>
              </li>

              <li>
                <a href="#">
                  Terms
                </a>
              </li>
              <li>
                <a href="#">
                  Privacy
                </a>
              </li>

              <li>
                <a href="#">
                  Sitemap
                </a>
              </li>

              <li>
                <a href="#">
                  Download
                </a>
              </li>

              <li>
                <a href="#">
                  Due Date Calculator
                </a>
              </li>

              <li>
                <a href="#">
                  Chinese Gender Predictor
                </a>
              </li>

              <li>
                <a href="#">
                  Ovulation Calculator
                </a>
              </li>

              <li>&nbsp;</li>
            </ul>
          </div>
        </section>
        
        <section className={styles.language}>
          <ul>
            <li>
              <a href="#" className={styles.selected}>
                English
              </a>
            </li>
            <li>
              <a href="#">
                Spanish
              </a>
            </li>
          </ul>
        </section>

        <section className={styles.copyright}>
          <div>&copy; {new Date().getFullYear()} Peanut</div>
        </section>
      </footer>
    </div>
  );
};

export default PageFooter;
