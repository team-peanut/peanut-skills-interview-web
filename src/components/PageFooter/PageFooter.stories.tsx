import React from "react";
import { ComponentStory } from "@storybook/react";

import PageFooter from "./PageFooter";

export default {
  title: "Components/PageFooter",
  component: PageFooter,

  parameters: {
    layout: "fullscreen",
    backgrounds: {
      default: "light",
    },
  },
};

const Template: ComponentStory<typeof PageFooter> = () => <PageFooter />;

export const Primary = Template.bind({});
