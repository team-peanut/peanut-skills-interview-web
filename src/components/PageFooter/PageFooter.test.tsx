import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";


import PageFooter from "./PageFooter";


describe("PageFooter", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  
  afterEach(cleanup);

  it("renders as expected", () => {
    const { asFragment } = render(<PageFooter />);
    expect(asFragment()).toMatchSnapshot();
  });

});
