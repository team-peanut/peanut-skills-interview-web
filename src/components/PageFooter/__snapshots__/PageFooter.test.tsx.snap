// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFooter renders as expected 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
    data-testid="page-footer"
  >
    <footer
      class="container"
    >
      <section
        class="install"
      >
        <button
          class="button"
        >
          <img
            class="buttonIOS"
            src="apple.svg"
          />
        </button>
        <button
          class="button"
        >
          <img
            class="buttonAndroid"
            src="robot.svg"
          />
        </button>
      </section>
      <section
        class="social"
      >
        <ul>
          <li>
            <a
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Instagram"
                height="22"
                src="instagram.svg"
                width="22"
              />
            </a>
          </li>
          <li>
            <a
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Facebook"
                height="22"
                src="facebook.svg"
                width="22"
              />
            </a>
          </li>
          <li>
            <a
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Twitter"
                height="22"
                src="twitter.svg"
                width="27"
              />
            </a>
          </li>
          <li>
            <a
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="TikTok"
                height="22"
                src="tiktok.svg"
                width="18"
              />
            </a>
          </li>
          <li>
            <a
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Pinterest"
                height="22"
                src="pinterest.svg"
                width="22"
              />
            </a>
          </li>
        </ul>
      </section>
      <section
        class="contacts"
      >
        <ul>
          <li>
            <a
              href="#"
            >
              Advertising Enquiries
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Press Enquiries
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Something else?
            </a>
          </li>
        </ul>
      </section>
      <section
        class="links"
      >
        <div>
          <ul>
            <li>
              <a
                href="#"
              >
                About us
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Blog
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Medical Review
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Guidelines
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Offers
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Start Her
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Shop
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Peanut News
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Careers
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Terms
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Privacy
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Sitemap
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Download
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Due Date Calculator
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Chinese Gender Predictor
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Ovulation Calculator
              </a>
            </li>
            <li>
               
            </li>
          </ul>
        </div>
      </section>
      <section
        class="language"
      >
        <ul>
          <li>
            <a
              class="selected"
              href="#"
            >
              English
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Spanish
            </a>
          </li>
        </ul>
      </section>
      <section
        class="copyright"
      >
        <div>
          © 2023 Peanut
        </div>
      </section>
    </footer>
  </div>
</DocumentFragment>
`;
